### PR TITLE
Snowflake destination : Enable DAT tests

### PR DESCRIPTION
**Pull request by DoNotPanicUA**
## What
The Snowflake destination fails the DAT tests. It means that we can't guarantee full data type compliance.

## How
Enable DAT tests and fix all errors.

## Recommended reading order
1. `SnowflakeDestinationIntegrationTest.java`
2. `SnowflakeTestDataComparator.java`
3. `SnowflakeTestSourceOperations.java`

### CODE REVIEW

**Review by sashaNeshcheret — APPROVED** [review]

File: airbyte-integrations/connectors/destination-snowflake/src/test-integration/java/io/airbyte/integrations/destination/snowflake/SnowflakeInsertDestinationAcceptanceTest.java [hunk 1]
```diff
@@ -118,29 +133,13 @@ protected List<JsonNode> retrieveNormalizedRecords(final TestDestinationEnv test
       throws Exception {
     final String tableName = NAME_TRANSFORMER.getIdentifier(streamName);
     final String schema = NAME_TRANSFORMER.getNamespace(namespace);
-    // Temporarily disabling the behavior of the ExtendedNameTransformer, see (issue #1785) so we don't
-    // use quoted names
-    // if (!tableName.startsWith("\"")) {
-    // // Currently, Normalization always quote tables identifiers
-    // tableName = "\"" + tableName + "\"";
-    // }
```
[the remark this answers is not in the document]
Comment (DoNotPanicUA) [reply airbyte-integrations/connectors/destination-snowflake/src/test-integration/java/io/airbyte/integrations/destination/snowflake/SnowflakeInsertDestinationAcceptanceTest.java:126]: It looks like an ancient artifact. Probably waiting for archaeologists.
Ok, let's do this!

**Pull request merged**